R/input-submit.R

```diff
@@ -0,0 +1,161 @@
+#' Create a textarea input control with explicit submission
+#'
+#' Creates a textarea input where users can enter multi-line text and submit
+#' their input using a dedicated button or keyboard shortcut. This control is
+#' ideal when you want to capture finalized input, rather than reacting to every
+#' keystroke, making it useful for chat boxes, comments, or other scenarios
+#' where users may compose and review their text before submitting.
+#'
+#' @param id The input ID.
+#' @param label The label to display above the input control. If `NULL`, no
+#'   label is displayed.
+#' @param placeholder The placeholder text to display when the input is empty.
+#'   This can be used to provide a hint or example of the expected input.
+#' @param value The initial input text. Note that, unlike [textAreaInput()],
+#'   this won't set a server-side value until the value is explicitly submitted.
+#' @param ... Currently not used.
+#' @param button A [tags] element to use for the submit button. It's recommended
+#'   that this be a [input_task_button()] since it will automatically provide a
+#'   busy indicator (and disable) until the next flush occurs. Note also that if
+#'   the submit button launches a [ExtendedTask], this button can also be bound
+#'   to the task ([bind_task_button()]) and/or manually updated for more
+#'   accurate progress reporting ([update_task_button()]).
+#' @param width Any valid CSS unit (e.g., `width="100%"`).
+#' @param submit_key A character string indicating what keyboard event should
+#'   trigger the submit button. The default is `enter`, which will submit the
+#'   input when the user presses the Enter/Return key. The `enter+modifier`
+#'   option will submit the input when the user presses the Enter key while
+#'   holding down Ctrl/Cmd.
+#'
+#' @return A textarea input control that can be added to a UI definition.
+#'
+#' @seealso [update_submit_textarea()], [input_task_button()]
+#'
+#' @examplesIf rlang::is_interactive()
+#'
+#' ui <- page_fluid(
+#'   input_submit_textarea("text", placeholder = "Enter some input..."),
+#'   verbatimTextOutput("value")
+#' )
+#' server <- function(input, output) {
+#'   output$value <- renderText({
+#'     req(input$text)
+#'     Sys.sleep(2)
+#'     paste("You entered:", input$text)
+#'   })
+#' }
+#' shinyApp(ui, server)
+#'
+#' @section Server value:
+#' A character string of the text input. The default value is `""` even if
+#' `value` is provided. The value will only be set/updated when the user submits
+#' the input by pressing the Enter key or clicking the submit button.
+#'
+#' @export
+input_submit_textarea <- function(
+  id,
+  label = NULL,
+  placeholder = NULL,
+  value = "",
+  ...,
+  button = NULL,
+  width = "min(680px, 100%)",
+  submit_key = c("enter", "enter+modifier")
+) {
+  rlang::check_installed("shiny", version = "1.11.1")
+  rlang::check_dots_empty()
```

Comment:
I just realized we're not making it easy to add classes or styles anywhere in this component. Suppose, e.g. I want a tighter border radius on the text input, or to adjust the font in the text input component.

Since users can provide the button component as an argument, it might be nice to allow attributes in `...` to be passed to `tags$textarea()`

Reply:
I ended up taking this in a direction that overlaps somewhat with the toolbar concept we discussed earlier ([e9ef2eb](https://github.com/rstudio/bslib/pull/1204/commits/e9ef2eba3037381e21bb27a856ad312a84c8d859)), so you can now do something like:

```r
input_submit_textarea(
      "text",
      "Your story",
      placeholder = "Once upon a time...",
      actionLink("about", "About")
    )
```

to get

<img width="274" height="157" alt="Screenshot 2025-10-07 at 5 12 10 PM" src="https://github.com/user-attachments/assets/15af2c4d-8e43-46f4-bf55-4fba741a25d0" />